zipkin-local.js

```diff
@@ -0,0 +1,129 @@
+"use strict";
+const _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
+const {
+  THEME_OPTIONS,
+} = require("gatsby-plugin-newrelic-test/utils/constants");
+const { getCiData } = require("./utils/getCiData");
+const ciAttributes = getCiData();
+exports.__esModule = true;
+exports.stop = exports.create = void 0;
+
+const _zipkin = _interopRequireDefault(require("zipkin"));
+
+const _zipkinTransportHttp = require("zipkin-transport-newrelic");
+
+const _zipkinJavascriptOpentracing = _interopRequireDefault(
+  require("zipkin-javascript-opentracing")
+);
+const _nodeFetch = _interopRequireDefault(require("node-fetch"));
+let logger;
+let recorder;
+/**
+ * Create and return an open-tracing compatible tracer. See
+ * https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.ts
+ */
+
+const create = () => {
+  logger = new _zipkinTransportHttp.HttpLogger({
+    // endpoint of local docker zipkin instance
+    endpoint: `https://${
+      THEME_OPTIONS.staging ? `staging-` : ``
+    }trace-api.newrelic.com/trace/v1`,
+    headers: {
+      "Api-Key": THEME_OPTIONS.NR_INGEST_KEY,
+      "Data-Format": "zipkin",
+      "Data-Format-Version": 2,
+      options: THEME_OPTIONS,
+      tags: {
+        ...THEME_OPTIONS.traces.tags,
+        ...ciAttributes,
+      },
+    },
+  });
+  recorder = new _zipkin.BatchRecorder({
+    logger,
+    // timeout = 60 hours, must be longer than site's build time
+    timeout: 60 * 60 * 60 * 1000000,
+  });
+  // console.log(recorder)
+  const tracer = new _zipkinJavascriptOpentracing.default({
+    localServiceName: THEME_OPTIONS.SITE_NAME,
+    serviceName: THEME_OPTIONS.SITE_NAME,
+    sampler: new _zipkin.sampler.CountingSampler(1),
+    traceId128Bit: true,
+    recorder,
+    kind: `client`,
+  });
+  return tracer;
+};
+exports.create = create;
+const _processQueue = async () => {
+  if (!THEME_OPTIONS.NR_INGEST_KEY || !THEME_OPTIONS.traces.collectTraces) {
+    return;
+  }
+  if (logger.queue.length > 0) {
```

Comment:
Since there's no `else` case, we might want to do an early return for the inverse condition:

```js
if (logger.queue.length <= 0) {
  return;
}
```

If we rewrite the logic in that way, it uncovers another area for improvement: grouping conditions in which we don't want to process the queue:

```js
const { INGEST_KEY, traces } = THEME_OPTIONS;

if (!NR_INGEST_KEY || !traces.collectTraces || logger.queue.length <= 0) {
  return;
}
```

Reply:
oh hello that would be much nicer